### ISSUE TITLE
Keras HDF5 model loading is broken

### ISSUE THREAD

**Issue by tpwrules** [NixOS/nixpkgs#147321]
### Describe the bug
Keras crashes trying to load HDF5 format models

### Steps To Reproduce
```
$ nix-shell -p 'python3.withPackages (p: with p; [ tensorflow_2 Keras ])' --run python3
Python 3.8.11 (default, Jun 28 2021, 10:57:31) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from keras.models import Sequential, load_model
>>> m = Sequential()
>>> m.build((1,))
>>> m.save("test.h5")
>>> load_model("test.h5")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/55rirzxfv37rsz5c13dh88hnj0apzhva-python3-3.8.11-env/lib/python3.8/site-packages/tensorflow/python/keras/saving/save.py", line 206, in load_model
    return hdf5_format.load_model_from_hdf5(filepath, custom_objects,
  File "/nix/store/55rirzxfv37rsz5c13dh88hnj0apzhva-python3-3.8.11-env/lib/python3.8/site-packages/tensorflow/python/keras/saving/hdf5_format.py", line 182, in load_model_from_hdf5
    model_config = json_utils.decode(model_config.decode('utf-8'))
AttributeError: 'str' object has no attribute 'decode'
>>> 
```

### Expected behavior
I expect models to load without crashing.

### Additional context
This is a known Keras bug due to an incompatibility with h5py >= 3.0.0. Solutions I have tried that worked include building both Keras and Tensorflow with h5py 2.10.0 (which must itself be built with the old HDF5 1.10), and patching Tensorflow as noted here: https://github.com/tensorflow/tensorflow/issues/44467#issuecomment-720042186 . Allegedly updating Tensorflow past 2.4 also works.

IMHO it would be best to update both Tensorflow and Keras to 2.7.0 and backport the updates to at least 21.11.

### Notify maintainers

cc @veprbl

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.0-90-generic, Ubuntu, 20.04.2 LTS (Focal Fossa)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20211006_53e4794`
 - channels(tpwatson): `"nixpkgs-21.05.4349.09650059d7f"`
 - nixpkgs: `/home/user/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by tpwrules** [comment]
This issue is fixed by Tensorflow and Keras 2.7.0.

**Comment by tpwrules** [comment]
This is also the same bug as #124599
